### PR TITLE
Build Fortran Libraries

### PR DESCRIPTION
**Pull request by ashao**
make lib now builds a Fortran library. For now this essentially
rebuilds all the code, but is necessary to avoid problems that
CMake has with Fortran dependencies when doing a parallel build
with multiple targets.

### CODE REVIEW

**Comment by ashao** [comment]
Here's a question for the group (because of the CI failing): do we want to build the Fortran libraries by default or should we add a `--build_fortran_client` as an option?

**Comment by billschereriii** [comment]
Looks like a lot of the build tests failed -- could you look into this please?

**Comment by ashao** [comment]
@billschereriii Yes, it's because it puts `enable_language(Fortran)` in the `CMakeLists.txt` and so all the Dockerfiles would need to be updated. I'm thinking that we might just setup the CMake build to include the Fortran client as a separate target

**Comment by billschereriii** [comment]
I think there is a split here. For CI testing, we should always have a Fortran compiler available and should be validating Fortran builds. On customer and development sites, however, we may not have one. We can't fail builds if a compiler isn't available, of course, so that's why I was suggesting adding the Fortran component to the library IFF the compiler is available.

**Comment by Spartee** [comment]
Most people don't have a Fortran complied pre installed. (I.e all our macs) 

We should not strictly require the presence of a Fortran compiler.

**Comment by ashao** [comment]
@mellis13 I like the object library as an intermediate solution. I _think_ that should work, so long as CMake respects the `.mod` files that Fortran generates. I'll prototype your solution and see how it goes.

**Comment by billschereriii** [comment]
> @mellis13 I like the object library as an intermediate solution. I _think_ that should work, so long as CMake respects the `.mod` files that Fortran generates. I'll prototype your solution and see how it goes.

@ashao How did this prototyping effort go?

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,10 +33,10 @@ option(BUILD_PYTHON "Build the python module" ON)
 set(CMAKE_BUILD_TYPE RELEASE)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_INSTALL_PREFIX ${CMAKE_SOURCE_DIR}/install)
+enable_language(Fortran)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```
[the remark this answers is not in the document]
Comment (ashao) [reply CMakeLists.txt]: Thanks for this catch. I was working off an old version of `develop` that still had it set as `hidden`. Pretty clear that it's one of the reasons why the tests were failing

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,17 +28,25 @@ project(SmartRedis)
...
+if (BUILD_FORTRAN)
+    enable_language(Fortran)
+endif()
 
 if (WERROR)
-    add_compile_options(-Wall -Werror)
+    if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+        add_compile_options(-Wall -Werror)
+    else()
+        message( WARNING "WERROR was specified, but the CMAKE compiler is not GCC")
```
Comment (billschereriii) [inline CMakeLists.txt:48]: Remove leading space before `WARNING`

**Pull request merged**